Helm/opensearch-dashboards/templates/deployment.yaml

```diff
@@ -0,0 +1,156 @@
+      # Currently some extra blocks accept strings
+      # to continue with backwards compatibility this is being kept
+      # whilst also allowing for yaml to be specified too.
+      initContainers:
+      {{- if eq "string" (printf "%T" .Values.extraInitContainers) }}
+{{ tpl .Values.extraInitContainers . | indent 6 }}
+      {{- else }}
+{{ toYaml .Values.extraInitContainers | indent 6 }}
+      {{- end }}
+      {{- end }}
+      containers:
+      - name: dashboards
+        securityContext:
+{{ toYaml .Values.securityContext | indent 10 }}
+        image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        env:
+          {{- if .Values.opensearchURL }}
+          - name: OPENSEARCH_URL
+            value: "{{ .Values.opensearchURL }}"
+          {{- else if .Values.opensearchHosts }}
+          - name: OPENSEARCH_HOSTS
+            value: "{{ .Values.opensearchHosts }}"
+          {{- end }}
+          - name: SERVER_HOST
+            value: "{{ .Values.serverHost }}"
+            {{- if .Values.opensearchAccount.secret }}
+            - name: OPENSEARCH_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: username
+            - name: OPENSEARCH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: password
+            {{- if and .Values.opensearchAccount.keyPassphrase.enabled }}
+            - name: KEY_PASSPHRASE
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: keypassphrase
+            # 32-character random string to be used as cookie password by security plugin
+            {{- end }}
+            - name: COOKIE_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
+                  key: cookie
+            {{- end }}
+{{- if .Values.extraEnvs }}
+{{ toYaml .Values.extraEnvs | indent 10 }}
+{{- end }}
+{{- if .Values.envFrom }}
+        envFrom:
+{{ toYaml .Values.envFrom | indent 10 }}
+{{- end }}
+        ports:
+        - containerPort: {{ .Values.httpPort }}
+{{- if .Values.lifecycle }}
+        lifecycle:
+{{ toYaml .Values.lifecycle | indent 10 }}
+{{- end }}
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+        volumeMounts:
+          {{- range .Values.secretMounts }}
+          - name: {{ .name }}
+            mountPath: {{ .path }}
+            {{- if .subPath }}
+            subPath: {{ .subPath }}
+            {{- end }}
+          {{- end }}
+          {{- range $path, $config := .Values.config }}
+          - name: config
+            mountPath: /usr/share/dashboards/config/{{ $path }}
+            subPath: {{ $path }}
+          {{- end -}}
```

Comment:
Can we copy how we now do this in the opensearch chart?

Reply:
I took reference from there only. [This](https://github.com/opensearch-project/opensearch-devops/blob/main/Helm/opensearch/templates/statefulset.yaml#L321) is how we did it for OpenSearch chart.

Reply:
@TheAlgo My bad, I'm thinking of something else. Resolving this.